mods/cnc/maps/gdi09/gdi09.lua

```diff
@@ -0,0 +1,163 @@
+Auto3 = { units = { ['e3'] = 2, ['e4'] = 2 }, waypoints = WaypointGroup1, delay = 50 }
+Auto4 = { units = { ['e1'] = 3, ['e4'] = 1 }, waypoints = WaypointGroup1, delay = 50 }
+Auto5 = { units = { ['ltnk'] = 1, ['bggy'] = 1 }, waypoints = WaypointGroup1, delay = 60 }
+Auto6 = { units = { ['bggy'] = 1 }, waypoints = WaypointGroup2, delay = 50 }
+Auto7 = { units = { ['ltnk'] = 1 }, waypoints = WaypointGroup2, delay = 50 }
+Auto8 = { units = { ['e4'] = 2, ['bggy'] = 1 }, waypoints = WaypointGroup4, delay = 0 }
+
+Patrols = 
+{ 
+	grd1 = { units = { ['e3'] = 3 }, waypoints = Patrol1Waypoints, wait = 40, initialWaypointPlacement = { 1 } },
+	grd2 = { units = { ['e1'] = 2, ['e3'] = 2, ['e4'] = 2 }, waypoints = Patrol2Waypoints, wait = 20, initialWaypointPlacement = { 4, 10, 1 } }
+}
+
+AutoAttackWaves = { Nod1, Nod2, Nod3, Nod4, Nod5, Auto1, Auto2, Auto3, Auto4, Auto5, Auto6, Auto7, Auto8 }
+
+StationaryGuards = { Actor174, Actor173, Actor182, Actor183, Actor184, Actor185, Actor186, Actor187 , Actor199, Actor200, Actor201, Actor202, Actor203, Actor204}
+
+StartStationaryGuards = function()
+	Utils.Do(StationaryGuards, function(unit)
+		if not unit.IsDead then
+			unit.Patrol( { unit.Location } , true, 20)
+		end
+	end)
+end
+
+SendWaves = function(counter, Waves)
+	if counter <= #Waves then
+		local team = Waves[counter]
+
+		for type, amount in pairs(team.units) do
+			MoveAndHunt(Utils.Take(amount, Nod.GetActorsByType(type)), team.waypoints)
+		end
+
+		Trigger.AfterDelay(DateTime.Seconds(team.delay), function() SendWaves(counter + 1, Waves) end)
+	end
+end
+
+StartPatrols = function()
+	for k, team in pairs(Patrols) do
+		local group = 1
+		for type, amount in pairs(team.units) do
+			for i=1,amount,1 do
+				Reinforcements.Reinforce(Nod, { type }, { team.waypoints[team.initialWaypointPlacement[group]] }, 0, function(unit)
+					ReplenishPatrolUnit(unit, handofnod, team.waypoints, team.wait)
+				end)
+			end
+			group = group + 1
+		end
+	end
+	Patrols = nil
+end
+
+ReplenishPatrolUnit = function(unit, building, waypoints, waitatwaypoint)
+	unit.Patrol(waypoints, true, waitatwaypoint)
+	Trigger.OnKilled(unit, function()
+		local queueUnit = { unit = { unit.Type }, atbuilding = { building }, waypoints = waypoints }
+		PatrolProductionQueue[#PatrolProductionQueue + 1] = queueUnit
+	end)
+end
+
+SendGDIReinforcements = function()
+	Media.PlaySpeechNotification(GDI, "Reinforce")
+	Reinforcements.ReinforceWithTransport(GDI, InsertionHelicopterType, GDIHeliReinfUnits, { GDIHeliEntryNorth.Location, GDIHeliLZ.Location }, { GDIHeliLZ.Location + CVec.New(20, 0) })
+end
+
+SendGDIReinforcementChinook = function()
+	Reinforcements.ReinforceWithTransport(GDI, 'tran', nil, { GDIHeliEntryNorth.Location, GDIHeliLZ.Location })
+end
+
+SpawnGunboat = function()
+	Media.PlaySpeechNotification(GDI, "Reinforce")
+	Actor.Create("boat", true, { Owner = GDI, Facing = 0, Location = waypoint23.Location })
```

Comment:
This appears to spawn the boat in the middle of the map and not at the map edge?

Reply:
True, back when I ported that mission I did't want to move a waypoint. I'll replace this with CPos.New(62,37)